packages/common/src/ContractInteractor.ts

```diff
@@ -178,6 +179,7 @@ export class ContractInteractor {
     this.versionManager = versionManager ?? new VersionsManager(gsnRuntimeVersion, gsnRequiredVersion)
     this.deployment = deployment
     this.provider = provider
+    this.signer = signer
```

Comment:
AFAIK, by having `this.signer = signer ?? provider.getSigner()`
you make it an optional parameter, and thus remove most of the needed changes below.